geth/transactions/factory.go

```diff
@@ -0,0 +1,188 @@
+package transactions
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/status-im/status-go/geth/account"
+	"github.com/status-im/status-go/geth/rpc"
+	"github.com/status-im/status-go/sign"
+)
+
+const (
+	// sendTransactionTimeout defines how many seconds to wait before returning result in sentTransaction().
+	sendTransactionTimeout = 300 * time.Second
+	rpcCallTimeout         = time.Minute
+
+	defaultGas = 90000
+)
+
+// Factory validates, signs and propagates transactions to the Ethereum network.
+type Factory struct {
+	signRequests      *sign.PendingRequests
+	ethTxClient       EthTransactor
+	completionTimeout time.Duration
+	rpcCallTimeout    time.Duration
+	networkID         uint64
+
+	addrLock   *AddrLocker
+	localNonce sync.Map
+	log        log.Logger
+}
+
+// NewFactory returns a new Manager.
+func NewFactory(signRequests *sign.PendingRequests) *Factory {
+	return &Factory{
+		signRequests:      signRequests,
+		addrLock:          &AddrLocker{},
+		completionTimeout: sendTransactionTimeout,
+		rpcCallTimeout:    rpcCallTimeout,
+		localNonce:        sync.Map{},
+		log:               log.New("package", "status-go/geth/transactions.Manager"),
+	}
+}
+
+// SetNetwork selects a correct network and RPC client.
+func (f *Factory) SetNetwork(networkID uint64, rpcClient *rpc.Client) {
+	f.log.Info("start Manager")
+	f.networkID = networkID
+	f.ethTxClient = NewEthTxClient(rpcClient)
+}
+
+// SendTransaction is an implementation of eth_sendTransaction. It queues the tx to the sign queue.
+func (f *Factory) SendTransaction(ctx context.Context, args SendTxArgs) (gethcommon.Hash, error) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	completeFunc := func(acc *account.SelectedExtKey) (gethcommon.Hash, error) {
+		return f.validateAndPropagate(acc, args)
+	}
+
+	request, err := f.signRequests.Add(ctx, args, completeFunc)
+
+	if err != nil {
+		return gethcommon.Hash{}, err
+	}
+
+	rst := f.signRequests.Wait(request.ID, f.completionTimeout)
+	return rst.Hash, rst.Error
+}
+
+// make sure that only account which created the tx can complete it
+func (f *Factory) validateAccount(args SendTxArgs, selectedAccount *account.SelectedExtKey) error {
+	if selectedAccount == nil {
+		return account.ErrNoAccountSelected
+	}
+
+	// make sure that only account which created the tx can complete it
```

Comment:
you probably don't need this comment anymore, it's already been set at the method's header.